scheduler/scheduler_unit_test.go

```diff
@@ -356,8 +367,51 @@ func (suite *SchedulerTestSuite) TestSchdulerDriverStopOK() {
 
 	suite.False(driver.Stopped())
 	suite.Equal(mesos.Status_DRIVER_RUNNING, driver.Status())
+	driver.connected = true // pretend that we're already registered
```

Comment:
Probably, by simulating the registration stuff. Doing things this way keep
the overall scope of the test a bit more focused. I don't think mutating
internal state like this is such a huge deal for a narrowly focused unit
test. If this were a larger component or integration test then I'd go with
the public method approach in order to transition the project properly.

On Sat, Jun 20, 2015 at 7:54 AM, Tomás Senart <notifications@github.com>
wrote:

> In scheduler/scheduler_unit_test.go
> <https://github.com/mesos/mesos-go/pull/125#discussion_r32884107>:
>
> > @@ -356,8 +367,51 @@ func (suite *SchedulerTestSuite) TestSchdulerDriverStopOK() {
> >
> >  	suite.False(driver.Stopped())
> >  	suite.Equal(mesos.Status_DRIVER_RUNNING, driver.Status())
> > +	driver.connected = true // pretend that we're already registered
>
> What I really wanted to ask was if there is a way to reach the same state
> using only public methods instead of short-cutting and mutating internal
> state.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/125/files#r32884107>.
>